examples/rust/sentiment/src/lib.rs

```diff
@@ -1,21 +1,34 @@
-wit_bindgen_rust::export!("sentiment.wit");
-
-struct Sentiment();
-
-impl sentiment::Sentiment for Sentiment {
-
-    fn sentiment(input: String) -> sentiment::PolarityScores {
-        lazy_static::lazy_static! {
-            static ref ANALYZER: vader_sentiment::SentimentIntensityAnalyzer<'static> =
-                vader_sentiment::SentimentIntensityAnalyzer::new();
-        }
-
-        let scores = ANALYZER.polarity_scores(input.as_str());
-        sentiment::PolarityScores {
-            compound: scores["compound"],
-            positive: scores["pos"],
-            negative: scores["neg"],
-            neutral: scores["neu"],
-        }
-    }
-}
+use serde::{ser::SerializeMap, Serialize, Serializer};
+
+wit_bindgen_rust::export!("sentiment.wit");
+
+impl Serialize for sentiment::PolarityScores {
```

Comment:
Shame that there isn't a way to add derives to WIT types

Reply:
Omg I agree 100%. I tried to hack this in a bunch of different ways. Even considered doing it in wit-bindgen, but it would require some annotation style idea in wit that may not be long-term desirable. Actually, what would likely be the most general solution is to have wit-bindgen support attaching onto previously defined structs rather than needing to generate all the types themselves. That might be a bit tricky, but would allow the user to just define the struct themself along with any additional derives needed.